XiEditor/CoreConnection.swift

```diff
@@ -84,14 +84,34 @@ class CoreConnection {
     func handleRaw(_ data: Data) {
         do {
             let json = try JSONSerialization.jsonObject(with: data, options: .allowFragments)
-            //print("got \(json)")
-            if !handleRpcResponse(json as AnyObject) {
+//            print("got \(json)")
+            handleRpc(json as Any)
+        } catch {
+            print("json error \(error.localizedDescription)")
+        }
+    }
+
+    /// handle a JSON RPC call. Determines whether it is a request, response or notifcation
+    /// and executes/responds accordingly
+    func handleRpc(_ json: Any) {
+        if let obj = json as? [String: Any], let index = obj["id"] as? Int {
+            if let result = obj["result"] { // is response
+                var callback: ((Any?) -> ())? = nil
```

Comment:
The `= nil` isn't necessary here, and doesn't seem to add to clarity, because it will always be assigned in the sync block below.